libs/ml/include/ml/model/model.hpp

```diff
@@ -320,8 +381,8 @@ void Model<TensorType>::TrainImplementation(DataType &loss, SizeType n_rounds)
 
   dataloader_ptr_->SetMode(dataloaders::DataLoaderMode::TRAIN);
 
-  loss_              = DataType{0};
-  DataType min_loss  = fetch::math::numeric_max<DataType>();
+  loss_ = DataType{0};
+  DataType min_loss;
```

Comment:
instead of declaring an uninstantiated value here, move the declaration to the first point of use a few lines later, i.e.:
DataType min_loss = loss_;